### PR TITLE
Added cmake options lists to install instructions

### PR DESCRIPTION
**Pull request by bhourahine**

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -259,6 +262,62 @@ In order to build DFTB+ carry out the following steps:
...
+
+Parallelism options
+^^^^^^^^^^^^^^^^^^^
+
++-----------------+--------+--------------------------------------------------+
+|Cmake option     | Default| Notes                                            |
++=================+========+==================================================+
+|-DWITH_OMP       | Y      |OpenMP parallelism enabled in the build.          |
++-----------------+--------+--------------------------------------------------+
+|-DWITH_MPI       | N      |MPI parallelism enanbled in the build.            |
```
Comment (vanderhe) [inline INSTALL.rst:313]:
```suggestion
|-DWITH_MPI       | N      |MPI parallelism enabled in the build.            |
```

File: INSTALL.rst [hunk 2]
```diff
@@ -300,6 +359,27 @@ Testing DFTB+
...
+|Option                  |Notes                                               |
++========================+====================================================+
+|-DTEST_MPI_PROCS        |Number of MPI processes used if testing an MPI build|
+|                        |with ctest.                                         |
++------------------------+----------------------------------------------------+
+|-DTEST_OMP_THREADS      |Number of openMP processes used if testing an openMP|
+|                        |build with ctest.                                   |
++------------------------+----------------------------------------------------+
+|-DTEST_RUNNER_TEMPLATE  |Modifies the DFTB+ invocation inside the test system|
+|                        |(for example if a specific mpi loader is required to|
```
Comment (vanderhe) [inline INSTALL.rst:379]:
```suggestion
|                        |(for example if a specific MPI loader is required to|
```

File: INSTALL.rst (hunk 1 above)
Comment (bhourahine) [inline INSTALL.rst:313]: Need an extra white space for the table, but yes should be changed.
```suggestion
|-DWITH_MPI       | N      |MPI parallelism enabled in the build.             |
```

Comment (vanderhe) [reply INSTALL.rst]: Sorry :D

**Review by vanderhe — APPROVED** [review]

**Pull request merged**